test/serialize.js

```diff
@@ -25,3 +25,26 @@ test('should not have null byte if no body', function() {
     assert.deepEqual(actual, expected);
 });
 
+test('should not pass undefined or null through', function() {
+    var msg = new Msgs.NewOrderSingle();
+
+    msg.OrderQty = undefined;
+    msg.Price = null;
+
+    var out = msg.serialize();
+
+    var expected = [
+        '8=FIX.4.2',
+        '9=57',
+        '35=D',
+        '52=undefined',
```

Comment:
I added the checks only for the body fields. The header fields, such as SenderCompID(49), which in practice should always be there, still pass through if they are undefined. (I kept this behavior, because in the other serialization test case they were explicitly let to pass through.)

The body fields, OrderQty(38) and Price(44), however do not appear in the expected value.